sdk/src/main/java/com/robotemi/sdk/Robot.kt

```diff
@@ -3367,6 +3404,75 @@ class Robot private constructor(private val context: Context) {
         return mapDataModel
     }
 
+    @WorkerThread
+    @Throws(IllegalArgumentException::class)
+    fun getFloorAndMapData(floorId: Int): Pair<Floor, MapDataModel>? {
+        if (floorId == 0) {
+            throw IllegalArgumentException("floorId must not be 0")
+        }
+
+        if (checkSelfPermission(Permission.MAP) == Permission.DENIED) {
+            Log.e(TAG, "getFloorAndMapData() - Permission denied for floorId=$floorId")
+            return null
+        }
```

Comment:
We don't need to check permission here. Since Robot class can be recompiled and all checks can be bypassed.